.github/workflows/build-http-apps-containers.yaml

```diff
@@ -0,0 +1,95 @@
+name: "Build HTTP Protocol Apps"
+
+on:
+  push:
+    branches:
+      - http-extensibility
+
+jobs:
+  device:
```

Comment:
it looks like most of the steps are identical for each job, consider using a matrix here to decrease duplication:

```yaml
jobs:
    strategy:
      matrix:
        component:
          - device
          - discovery
          - grpc-broker
          - grpc-client
```

Reply:
I'm new to GitHub Actions. Will look into this.

Reply:
Thanks for the pointer!

Simplified but not as ideally as it could be because I have Dockerfile w/ period e.g. `grpc.broker` but repository with hyphen e.g. `grpc-broker`. Yes, I could do some renames but periods in repository names aren't permitted and I've a sense (!?) hyphenated filenames can be problematic.

Fixed: https://github.com/deislabs/akri/pull/135/commits/f8d9de76067cf547cdf10a20c89c060f996f9b0e

Reply:
you could create several properties per matrix item:

```yaml
    strategy:
      matrix:
        include:
          - { component: device,   dockerfile: device,  container: http-apps-device }
          - { component: discovery,   dockerfile: discovery,  container: http-apps-discovery }
          - { component: grpc-broker,   dockerfile: grpc.broker,  container: http-apps-grpc-broker }
          - { component: grpc-client,   dockerfile: grpc.client,  container: http-apps-grpc-client }
```

Reply:
here's an example: https://github.com/chrisjbillington/zprocess/blob/d925a328ef6de3fd17e5c5a2abe63ae119740778/.github/workflows/build-and-release.yaml

Reply:
Oh, neat, thanks! :-)

I'll update it tomorrow.

Headed out to the dog park soon.

Reply:
Fixed: https://github.com/deislabs/akri/pull/135/commits/07c92974c55f6eee7bf96ba322643c51b5803085

Now I have to refine my other GitHub Actions elsewhere :-)